pyext/src/output.py

```diff
@@ -692,59 +692,148 @@ def write_stats2(self):
 
-        # get the keys from the first line
-        for line in f.readlines():
-            d = ast.literal_eval(line)
-            self.klist = list(d.keys())
-            # check if it is a stat2 file
-            if "STAT2HEADER" in self.klist:
-                self.isstat2 = True
-                for k in self.klist:
-                    if "STAT2HEADER" in str(k):
-                        # if print_header: print k, d[k]
-                        del d[k]
-                stat2_dict = d
-                # get the list of keys sorted by value
-                kkeys = [k[0]
-                         for k in sorted(stat2_dict.items(), key=operator.itemgetter(1))]
-                self.klist = [k[1]
-                              for k in sorted(stat2_dict.items(), key=operator.itemgetter(1))]
-                self.invstat2_dict = {}
-                for k in kkeys:
-                    self.invstat2_dict.update({stat2_dict[k]: k})
-            else:
-                IMP.handle_use_deprecated("statfile v1 is deprecated. "
-                                          "Please convert to statfile v2.\n")
-                self.isstat1 = True
-                self.klist.sort()
+        self.exp_dict={} # Store all experimental header details here.
+
+        # get the {CategoryID:Category} pairs from the first line
+        line = f.readline()
+
+        #Store these keys in a dictionary. Example pair: {109 :'Total_Score'}
+        self.dict = self.parse_line(line, header=True)
+        #self.dict = ast.literal_eval(line)
+
+        self.klist = list(self.dict.keys())
+
+        if "STAT2HEADER" in self.klist:
+            self.isstat2 = True
+            for k in self.klist:
+                if "STAT2HEADER" in str(k):
+                    # if print_header: print k, d[k]
+                    self.exp_dict[k]=self.dict[k]
+                    del self.dict[k]
+
+            # get the list of keys sorted by value
+            kkeys = [k[0]
+                    for k in sorted(self.dict.items(), key=operator.itemgetter(1))]
+            self.klist = [k[1]
+                    for k in sorted(self.dict.items(), key=operator.itemgetter(1))]
+            self.inv_dict = {}
+
+            for k in kkeys:
+                self.inv_dict.update({self.dict[k]: k})
+        else:
+            print("WARNING: statfile v1 is deprecated.  Please convert to statfile v2")
+            self.isstat1 = True
+            self.klist.sort()
+            # For v1, no need to map from ids to field names, so just make
+            # a dumb one-to-one mapping so as not to confuse v2 code
+            self.dict = {}
+            self.inv_dict = {}
+            for k in self.klist:
+                self.dict[k] = k
+                self.inv_dict[k] = k
+        
 
-            break
         f.close()
 
+
+    def parse_line(self, line, header=False):
+        # Parses a line and returns a dictionary of key:value pairs
```

Comment:
Use a docstring (`"foo"`) rather than comment (`#foo`) here so it gets picked up by doxygen. (If you don't want it to be part of the public interface - which requires documentation - make it a private function by calling it `_parse_line` rather than `parse_line`.)